core/keyboard_nav/ast_node.js

```diff
@@ -412,29 +415,30 @@ Blockly.ASTNode.prototype.findPrevForField_ = function() {
 /**
  * Navigate between stacks of blocks on the workspace.
  * @param {boolean} forward True to go forward. False to go backwards.
- * @return {Blockly.ASTNode} The first block of the next stack or null if there
+ * @return {ASTNode} The first block of the next stack or null if there
  * are no blocks on the workspace.
  * @private
  */
-Blockly.ASTNode.prototype.navigateBetweenStacks_ = function(forward) {
+ASTNode.prototype.navigateBetweenStacks_ = function(forward) {
   var curLocation = this.getLocation();
   if (curLocation.getSourceBlock) {
-    curLocation = /** @type {!Blockly.IASTNodeLocationWithBlock} */ (
-      curLocation).getSourceBlock();
+    curLocation = /** @type {!IASTNodeLocationWithBlock} */ (curLocation)
+                      .getSourceBlock();
   }
   if (!curLocation || !curLocation.workspace) {
     return null;
   }
-  var curRoot = curLocation.getRootBlock();
-  var topBlocks = curRoot.workspace.getTopBlocks(true);
-  for (var i = 0, topBlock; (topBlock = topBlocks[i]); i++) {
+  const curRoot = curLocation.getRootBlock();
+  const topBlocks = curRoot.workspace.getTopBlocks(true);
+  let i = 0, topBlock;
+  for (; (topBlock = topBlocks[i]); i++) {
```

Comment:
Is this what it's supposed to be? I know this technically works, but this is really weird looking to me though and I thought `for (let i = 0, topBlock; topBlock = topBlocks[i]; i++)` was supposed to work? (there are several like this in this file)

Reply:
This is what my IDE automatically updated. I'll look into it.

Reply:
The IDE is probably worried that `i` or `topBlock` might be used after the end of the loop body—which is possible with a `var` (function scoped), but not with `let` (scoped only to the loop).

A bit disappointing it doesn't actually check, though.